### ISSUE TITLE
Clarify wallet listing requirements for code being open source / publicly available

### ISSUE THREAD

**Issue by wbnns** [bitcoin-dot-org/bitcoin.org#3032]
We noted that at least a couple of wallets listed on the site do not have code that is open source and/or publicly available:

+ [BitGo](https://bitcoin.org/en/wallets/web/bitgo/) - no link to a code repository is present in the [wallet metadata](https://raw.githubusercontent.com/bitcoin-dot-org/bitcoin.org/master/_wallets/bitgo.md).
+ [BTC.com](https://wallet.btc.com) - [code repository](https://github.com/blocktrail/blocktrail-webwallet) links to [Blocktrail](https://www.blocktrail.com), which has a notice that they have ceased operations. The wallet at wallet.btc.com doesn't appear to share the same codebase. For example, it has a buy bitcoin feature which isn't present in the GitHub repository for the aforementioned web wallet by Blocktrail.

Since wallet providers are being instructed that their codebase must be publicly available, we wanted to clarify how this requirement should pertain to existing wallets on the site.

**Comment by crwatkins** [comment]
> Since wallet providers are being instructed that their codebase must be publicly available, we wanted to clarify how this requirement should pertain to existing wallets on the site.

I can comment more on this later, but I believe that we expect to have source code links for web wallet code that handles private keys.  On the two issues mentioned, I would like to clarify:

>BTC.com - code repository links to Blocktrail, which has a notice that they have ceased operations. The wallet at wallet.btc.com doesn't appear to share the same codebase. For example, it has a buy bitcoin feature which isn't present in the GitHub repository for the aforementioned web wallet by Blocktrail.

We have a source code link in the web btc.com wallet listing.  Are you stating that the code we link to is not the current code being used for the wallet?

> BitGo - no link to a code repository is present in the wallet metadata.

I believe "we" (i.e. not BitGo) removed the source link when we dropped the chrome extension from the BitGo listing.  We should ask for the correct link.

**Comment by wbnns** [comment]
@crwatkins 

> We have a source code link in the web btc.com wallet listing. Are you stating that the code we link to is not the current code being used for the wallet?

Yes, the wallet at wallet.btc.com doesn't appear to share the same codebase as the source code link. For example, it has a buy bitcoin feature which doesn't seem to be present in the [GitHub repository](https://github.com/blocktrail/blocktrail-webwallet).

**Comment by crwatkins** [comment]
> Since wallet providers are being instructed that their codebase must be publicly available, we wanted to clarify how this requirement should pertain to existing wallets on the site.

All discussions that I know of on this forum in the last few years has concluded that public source code is desirable and required for listing.

In the "early days" I believe there may have been some assumptions that web wallets were intrinsically open source, but with advances in JavaScript and HTML compression, that's no longer true, if it ever was.

**Comment by crwatkins** [comment]
@mbelshe @bpdavenport @AlejandroDeLaTorre  Could you confirm that source code links are current in your wallet listings?  Thanks!

**Comment by AlejandroDeLaTorre** [comment]
I have pinged the relevant parties. I no longer am with BTC.com

**Comment by mbelshe** [comment]
Hey Craig - sorry for the delays.  Yes we can check this out.

Mike

**Comment by lsgrep** [comment]
Hi, I am a developer from BTC.com and our wallet is fully open source. 

@wbnns  Hi , you could check out buy bitcoin from here https://github.com/blocktrail/blocktrail-webwallet/blob/957fd8f4076fa3b4fe2732d9cd0337cc985c5570/src/js/modules/wallet/controllers/buy-btc-broker/buy-btc-broker.tpl.html

**Comment by bencxr** [comment]
hey @crwatkins, we have open-sourced our wallet library at: https://github.com/bitgo/bitgojs
Happy to make a PR if you'd like.

**Comment by crwatkins** [comment]
@bencxr Thanks.   A PR for your entry would be great.  Let us know if you have any questions.

**Comment by maxwellmons** [comment]
Greetings @crwatkins and @wbnns and BitGo team. That isn't BitGo's source code for the web wallet? It is an API client library?

**Comment by bencxr** [comment]
> Greetings @crwatkins and @wbnns and BitGo team. That isn't BitGo's source code for the web wallet? It is an API client library?

Hi @maxwellmons, that's correct. We use the library within the open source clients on github. We do not have all the website code public / on github.

**Comment by wbnns** [comment]
@paullinator 

Hello, during a recent review, we were unable to locate the source code repositories for Edge for iOS and Edge for Android. Could you please provide us with updated links?
